Modules/Sources/PointOfSale/Environment/POSEnvironmentKeys.swift

```
@@ -25,9 +29,79 @@ private struct DefaultPOSAnalytics: POSAnalyticsProviding {
     }
 }
 
+/// Environment key for POS currency settings
+public struct POSCurrencySettingsKey: EnvironmentKey {
+    public static let defaultValue: CurrencySettings = CurrencySettings()
+}
+
+/// Environment key for POS stores service
+public struct POSStoresKey: EnvironmentKey {
+    public static let defaultValue: POSStoresProviding = DefaultPOSStores()
+}
+
+/// Environment key for POS feature flags service
+public struct POSFeatureFlagsKey: EnvironmentKey {
+    public static let defaultValue: POSFeatureFlagProviding = DefaultPOSFeatureFlags()
+}
+
+/// Environment key for POS session manager
+public struct POSSessionManagerKey: EnvironmentKey {
+    public static let defaultValue: POSSessionManagerProviding = DefaultPOSSessionManager()
+}
+
+/// Environment key for POS connectivity
+public struct POSConnectivityKey: EnvironmentKey {
+    public static let defaultValue: ConnectivityObserver = DefaultPOSConnectivity()
+}
+
+// Default implementations for testing/previews
+private struct DefaultPOSStores: POSStoresProviding {
```

nit: since we also have `POSPreviewServices` that includes individual dependencies for previews, how about removing the default dependencies here if the purpose is for SwiftUI previews? Previews that need it can initialize `POSPreviewServices` and set the required environment keys. If we miss setting an environment key, having the app crash would make it more discoverable than some features silently failing IMO.

---

I don't know if we can avoid setting a default value, but I will check. What we could do is to reuse these Default stores within the Preview service. 

>  If we miss setting an environment key, having the app crash would make it more discoverable than some features silently failing IMO.

That's true, but it's also unlikely that we forget, given that we inject it at the entry point. When we add a new dependency when developing a feature, we will see that something is not working as it should.